Microsoft.Toolkit.Uwp.UI.Controls/HeaderedContentControl/HeaderedContentControl.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Provides the base implementation for all controls that contain single content and have a header.
+    /// </summary>
+    public class HeaderedContentControl : ContentControl
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HeaderedContentControl"/> class.
+        /// </summary>
+        public HeaderedContentControl()
+        {
+            DefaultStyleKey = typeof(HeaderedContentControl);
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Header"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty HeaderProperty = DependencyProperty.Register(
+            "Header",
+            typeof(object),
+            typeof(HeaderedContentControl),
+            new PropertyMetadata(null));
+
+        /// <summary>
+        /// Identifies the <see cref="HeaderTemplate"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty HeaderTemplateProperty = DependencyProperty.Register(
+            "HeaderTemplate",
+            typeof(DataTemplate),
+            typeof(HeaderedContentControl),
+            new PropertyMetadata(null));
+
+        /// <summary>
+        /// Gets or sets the data used for the header of each control.
+        /// </summary>
+        public object Header
+        {
+            get { return (object)GetValue(HeaderProperty); }
+            set { SetValue(HeaderProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the template used to display the content of the control's header.
+        /// </summary>
+        public DataTemplate HeaderTemplate
+        {
+            get { return (DataTemplate)GetValue(HeaderTemplateProperty); }
+            set { SetValue(HeaderTemplateProperty, value); }
+        }
+
+        /// <summary>
+        /// Called when the <see cref="Header"/> property changes.
+        /// </summary>
+        /// <param name="oldValue">The old value of the <see cref="Header"/> property.</param>
+        /// <param name="newValue">The new value of the <see cref="Header"/> property.</param>
+        protected virtual void OnHeaderChanged(object oldValue, object newValue)
```

Comment:
Is this needed?

Reply:
Following [API from WPF](https://msdn.microsoft.com/en-us/library/system.windows.controls.headeredcontentcontrol.onheaderchanged(v=vs.110).aspx)